services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,42 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
-  return ERR_CODE_SUCCESS;
+  if (xQueueSend(thermalMgrQueueHandle, event, portMAX_DELAY) == pdTRUE) return ERR_CODE_SUCCESS;
+  return ERR_CODE_QUEUE_FULL;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  //send an event everytime this handler is called
+  thermal_mgr_event_t event;
+  event.type = THERMAL_MGR_EVENT_OS_INTERRUPT;
+
+  thermalMgrSendEvent(&event);
+
+  
+
 }
 
+
 static void thermalMgr(void *pvParameters) {
+  thermal_mgr_event_t event;
+  float temp;
   /* Implement this task */
   while (1) {
     
+    if ((thermalMgrQueueHandle != NULL)){
+      if (xQueueReceive(thermalMgrQueueHandle, &event, portMAX_DELAY) == pdTRUE) {
+        readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp);
```

Comment:
check event.type before doing this, since we shouldn't be doing this if the event is invalid

Reply:
also check return value to make sure it was successful